scripts/qa_crowbarsetup.sh

```diff
@@ -5038,10 +5039,147 @@ function ping_fips
     done
 }
 
+function get_stack_fips
+{
+    local stack=$1
+    local fip_file=$2
+    local all_fips=$(mktemp)
+    local fip_uuids=$(openstack stack resource list $stack -f value -c physical_resource_id --filter type=OS::Neutron::FloatingIP)
+    openstack floating ip list -f value > $all_fips
+
+    for uuid in $fip_uuids
+        do
+        echo $(awk "\$1 ~ /${uuid}/ { print \$2 }" $all_fips) >> $fip_file
+        done
+
+    rm $all_fips
+}
+
+
+function ping_stack_fips
+{
+    . .openrc
+
+    local stack=$1
+    local fips=$(mktemp)
+
+    get_stack_fips $stack $fips
+
+    for fip in $(cat $fips)
+        do
+        ping -c 1 -w 60 $fip || complain 120 "cannot reach stack $stack floating IP $fip."
+        done
+
+    for uuid in $fip_uuids
```

Comment:
is this second loop needed? `$fip_uuids` is `local` so it's probably not even visible here.

Reply:
No it isn't - that's a leftover from copying things around between functions. Removed.